gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/config/WUMLListenerTest.java

```diff
@@ -41,17 +46,108 @@ public void setup() {
         InboundEPProviderRegistry.getInstance().registerInboundEPProvider(new TestProvider());
         OutboundEPProviderRegistry.getInstance().registerOutboundEPProvider(new TestOutboundEPProvider());
     }
+
     @Ignore
     @Test()
     public void testListernerConstantsIflow() {
         Assert.assertTrue((parseIflow("/integration-flows/constants.iflow")));
     }
+
     @Ignore
     @Test
     public void testListernerVariableAssignmentIflow() {
         Assert.assertTrue((parseIflow("/integration-flows/variable-assignment.iflow")));
     }
 
+    @Test
+    public void testPassThroughObjectModelCreation() {
+        String iFlowResource = "/integration-flows/passthrough.xyz";
+        String integrationName = "passthrough";
+        String resourceName = "passthrough";
+        Boolean asExpected;
+
+        Assert.assertTrue("Error while deploying config :" + iFlowResource, deployArtifacts(iFlowResource));
+
+        Integration sampleIntegration = IntegrationConfigRegistry.getInstance().getIntegrationConfig(integrationName);
+        MediatorCollection generatedMediatorCollection = sampleIntegration.getResource(resourceName).getDefaultWorker()
+                .getMediators();
+        List<Mediator> mediatorList = generatedMediatorCollection.getMediators();
+
+        // Evaluate the generated mediator collection
+        if (mediatorList.size() == 3) {
```

Comment:
Its 3 because according to the sample, there are three mediators. Its completely depend on the sample NEL config that I have used "passthrough.xyz". I will add a brief comment on this,
Thanks,